app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+from typing import Any
+
+
+def cache(func: Callable) -> Callable:
+    cached_results = {}
+
+    def wrapper(*args) -> Any:
+        key = args
```

Comment:
Not really needed to create new variable here
Also, you can name the parameters in a function definition as you like, but it's a common convention to use `*args`. So, `args` will fit good here 👍